leaflet-treering.js

```diff
@@ -348,19 +348,77 @@ function MeasurementData (dataObject) {
    * @function insertPoint
    */
   MeasurementData.prototype.insertPoint = function(latLng, hasLatewood) {
-    var i = 0;
-    while (this.points[i] != undefined &&
-        this.points[i].latLng.lng < latLng.lng) {
-      i++;
-    }
+    var disList = [];
+
+    /**
+    * calculate the distance between 2 points
+    * @function distanceCalc
+    * @param {first point.latLng} pointA
+    * @param {second point.latLng} pointB
+    */
+    function distanceCalc (pointA, pointB) {
+      return Math.sqrt(Math.pow((pointB.lng - pointA.lng), 2) +
+                       Math.pow((pointB.lat - pointA.lat), 2));
+    };
+
+    // finds point with smallest abs. distance
+    for (i = 0; i < this.points.length; i++) {
+      var distance = Number.MAX_SAFE_INTEGER;
+      if (this.points[i] && this.points[i].latLng) {
+         var currentPoint = this.points[i].latLng;
+         distance = distanceCalc(currentPoint, latLng);
+      disList.push(distance);
+    };
+
+    var minDistance = Math.min(...disList);
+    i = disList.indexOf(minDistance)
+
     if (this.points[i] == null) {
-      alert('New point must be within existing points.' +
-          'Use the create toolbar to add new points to the series.');
+      alert('New point must be within existing points. Use the create toolbar to add new points to the series.');
       return;
     }
 
+    // define 3 points: i, i - 1, i + 1
+    var iPoint = this.points[i].latLng;
+
+    if (this.points[i - 1].start) {
```

Comment:
no guarantee this.points[I-1] exists, so calling .start on it could halt execution. Safe way is 
if(this.points[I-1] && this.points[I-1].start) {}

Reply:
I guess you don't actually care about the .start anyways?

Reply:
The .start make sure the inserted point doesn't connect to previous measurement path.